packages/payload/src/queues/operations/runJobs/runJob/getRunTaskFunction.ts

```diff
@@ -0,0 +1,309 @@
+import type { PayloadRequest } from '../../../../types/index.js'
+import type {
+  RetryConfig,
+  RunInlineTaskFunction,
+  RunTaskFunction,
+  RunTaskFunctions,
+  TaskConfig,
+  TaskHandler,
+  TaskHandlerResult,
+  TaskType,
+} from '../../../config/types/taskTypes.js'
+import type {
+  BaseJob,
+  RunningJob,
+  SingleTaskStatus,
+  WorkflowConfig,
+  WorkflowTypes,
+} from '../../../config/types/workflowTypes.js'
+import type { UpdateJobFunction } from './getUpdateJobFunction.js'
+
+import { calculateBackoffWaitUntil } from './calculateBackoffWaitUntil.js'
+import { importHandlerPath } from './importHandlerPath.js'
+
+// Helper object type to force being passed by reference
+export type RunTaskFunctionState = {
+  reachedMaxRetries: boolean
+}
+
+async function getTaskHandlerFromConfig(taskConfig: TaskConfig<string>) {
+  let handler: TaskHandler<TaskType>
+
+  if (typeof taskConfig.handler === 'function') {
+    handler = taskConfig.handler
+  } else {
+    handler = await importHandlerPath<TaskHandler<TaskType>>(taskConfig.handler)
+  }
+  return handler
+}
+
+export async function handleTaskFailed({
+  error,
+  executedAt,
+  input,
+  job,
+  maxRetries,
+  output,
+  req,
+  retriesConfig,
+  runnerOutput,
+  state,
+  taskConfig,
+  taskID,
+  taskSlug,
+  taskStatus,
+  updateJob,
+}: {
+  error?: Error
+  executedAt: Date
+  input: object
+  job: BaseJob
+  maxRetries: number
+  output: object
+  req: PayloadRequest
+  retriesConfig: number | RetryConfig
+  runnerOutput?: TaskHandlerResult<string>
+  state: RunTaskFunctionState
+  taskConfig?: TaskConfig<string>
+  taskID: string
+  taskSlug: string
+  taskStatus: null | SingleTaskStatus<string>
+  updateJob: UpdateJobFunction
+}): Promise<never> {
+  req.payload.logger.error({ err: error, job, msg: 'Error running task', taskSlug })
```

Comment:
I don't think we should log the entire job because it is very verbose and also it includes the full `input` could have sensitive info.
Ideally this level of logging would only be enabled with a flag for `debug: true` I think.